core/trino-spi/src/main/java/io/trino/spi/type/Type.java

```diff
@@ -174,6 +174,14 @@ default TypeOperatorDeclaration getTypeOperatorDeclaration(TypeOperators typeOpe
         return Optional.empty();
     }
 
+    /**
+     * Returns the nearest greater value (if exists).
+     */
+    default Optional<?> getNearestGreaterValue(Object value)
```

Comment:
This is potentially also useful for TIME_SECONDS, TIME_MILLIS, DATE, TIMESTAMP_SECONDS, TIMESTAMP_MILLIS, ShortDecimalType

Reply:
Please let me know if https://github.com/trinodb/trino/pull/9868/commits/9da363da19f4588fdaf0ef416fc9e5d4ed3408fe looks good - I'll be happy to implement it for more types.

Reply:
Forgot to mention it earlier, you can implement this now for the above types too

Reply:
Thanks - I have added 7abc9c3ac9c2b179ed9a93da144ad217fdc8be72, please let me know if this is the correct way to add support for those types...